pkg/capabilities/triggers/cron_trigger.go

```diff
@@ -0,0 +1,221 @@
+		scheduledExecutionTimeFormatted := scheduledExecutionTime.Format(time.RFC3339)
+		hash := sha256.Sum256([]byte(scheduledExecutionTimeFormatted))
+		triggerEventID := hex.EncodeToString(hash[:])
+
+		currentTime := time.Now()
+
+		// cts.lggr.Debugw
+		fmt.Println("process", "scheduledExecTime", scheduledExecutionTime.Format(time.RFC3339Nano), "actualExecTime", currentTime.UTC().Format(time.RFC3339Nano), "triggerEventId", triggerEventID)
+
+		response, errr := wrapCronTriggerEvent(triggerEventID, currentTime, scheduledExecutionTimeFormatted)
+		if errr != nil {
+			cts.lggr.Errorw("error wrapping trigger event", "err", errr)
+		}
+
+		select {
+		case callbackCh <- response:
+		default:
+			cts.lggr.Errorw("channel full, dropping event", "triggerID", req.TriggerID, "eventID", triggerEventID, "capabilityResponse", response)
+		}
+	})
+	job, err = cts.scheduler.NewJob(jobDef, task)
+
+	cts.triggers[req.TriggerID] = cronTrigger{
+		ch:  callbackCh,
+		job: job,
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	cts.lggr.Debugw("RegisterTrigger", "triggerId", req.TriggerID, "jobId", job.ID())
+	return callbackCh, nil
+}
+
+func (cts *CronTriggerService) UnregisterTrigger(ctx context.Context, req capabilities.TriggerRegistrationRequest) error {
+	cts.mu.Lock()
+	defer cts.mu.Unlock()
+
+	job, ok := cts.triggers[req.TriggerID]
+	if !ok {
+		return fmt.Errorf("triggerId %s not found", triggerID)
+	}
+
+	jobID := job.job.ID()
+
+	// Remove job from scheduler
+	err := cts.scheduler.RemoveJob(jobID)
+	if err != nil {
+		return fmt.Errorf("UnregisterTrigger failed to remove job from scheduler: %s", err)
+	}
+
+	// Close callback channel
+	close(job.ch)
+	// Remove from triggers context
+	delete(cts.triggers, req.TriggerID)
+
+	cts.lggr.Debugw("UnregisterTrigger", "triggerId", triggerID, "jobId", jobID)
+
+	return nil
+}
+
+// Start the service.
+func (cts *CronTriggerService) Start(ctx context.Context) error {
+	if cts.scheduler == nil {
+		return errors.New("no scheduler initialized")
+	}
+	cts.scheduler.Start()
+
+	cts.lggr.Info(cts.Name() + " started")
+
+	return nil
+}
+
+// Close stops the Service.
+// Invariants: After this call the Service cannot be started
+// again, new Service will need to be built to do so.
+func (cts *CronTriggerService) Close() error {
+	err := cts.scheduler.Shutdown()
+	if err != nil {
+		return fmt.Errorf("scheduler shutdown encountered a problem: %s", err)
+	}
+
+	cts.lggr.Info(cts.Name() + " closed")
+
+	return nil
+}
+
+func (cts *CronTriggerService) Ready() error {
+	return nil
+}
+
+func (cts *CronTriggerService) HealthReport() map[string]error {
+	return map[string]error{cts.Name(): nil}
+}
+
+func (cts *CronTriggerService) Name() string {
+	return "CronTriggerService"
+}
+
+func wrapCronTriggerEvent(eventID string, timestamp time.Time, scheduledExecutionTime string) (capabilities.TriggerRegistrationResponse, error) {
+	payload, err := values.Wrap(CronTriggerPayload{ScheduledExecutionTime: scheduledExecutionTime})
```

Comment:
@justinkaseman Have you tested that this works? I don't think we have a time.Time type in the values library

Reply:
`scheduledExecutionTime` probably should have been named `scheduledExecutionTimeString`

Collapsed `wrapCronTriggerEvent` away since it's fairly small now. Hopefully clears makes it more clear.